### PR TITLE
SWC-6759 - fix xss vector

### PR DESCRIPTION
**Pull request by nickgros**

### CODE REVIEW

File: packages/synapse-react-client/src/utils/functions/SanitizeHtmlUtils.ts [hunk 1]
```diff
@@ -93,18 +93,20 @@ export const xssOptions: IFilterXSSOptions = {
     }
     return undefined
   },
-  safeAttrValue: function (tag, name, value) {
+  safeAttrValue: function (tag, name, value, cssFilter) {
+    // Apply default safeAttrValue filtering:
+    value = safeAttrValue(tag, name, value, cssFilter)
```
Comment (nickgros) [inline packages/synapse-react-client/src/utils/functions/SanitizeHtmlUtils.ts:98]: The default `safeAttrValue` will handle the vulnerable case

File: packages/synapse-react-client/src/utils/functions/SanitizeHtmlUtils.test.ts [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+  })
+  test('Prevents XSS through href', () => {
+    const script = 'javascript:alert(1)'
+    const sanitized = xss(`<a href="${script}">foo</span>`, xssOptions)
+    const html = createHTML(sanitized)
+    const anchor = html.querySelector('a')!
+    expect(anchor).not.toBeNull()
+    // Verify script was removed
+    expect(anchor.getAttribute('href')).toEqual('')
+  })
```
Comment (nickgros) [inline packages/synapse-react-client/src/utils/functions/SanitizeHtmlUtils.test.ts:33]: This test verifies the newly discovered vulnerability. I verified that it failed before the changes

File: packages/synapse-react-client/src/utils/functions/SanitizeHtmlUtils.ts [hunk 3]
```diff
@@ -1,4 +1,4 @@
-import { IFilterXSSOptions } from 'xss'
+import { escapeAttrValue, IFilterXSSOptions, safeAttrValue } from 'xss'
 
 // PORTALS-1450: including 'style' in the allow-list will cause string values to come through, which crashes the app when used (because it uses jsx).
 export const xssOptions: IFilterXSSOptions = {
```
Comment (nickgros) [inline packages/synapse-react-client/src/utils/functions/SanitizeHtmlUtils.ts:4]: I'll make a ticket to expose this and have just one single XSS sanitization method

File: packages/synapse-react-client/src/utils/functions/SanitizeHtmlUtils.test.ts [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+    const span = html.querySelector('span')!
+    expect(span).not.toBeNull()
+    expect(span.getAttribute('onload')).toBeNull()
+  })
+  test('Prevents XSS through onclick', () => {
+    const sanitized = xss("<span onclick='alert('XSS')'>foo</span>", xssOptions)
+    const html = createHTML(sanitized)
+    const span = html.querySelector('span')!
+    expect(span).not.toBeNull()
+    expect(span.getAttribute('onload')).toBeNull()
```
Comment (hallieswan) [inline packages/synapse-react-client/src/utils/functions/SanitizeHtmlUtils.test.ts:23]: Should this be `onclick`?

**Review by hallieswan — APPROVED** [review]
Looks good to me! Just one small comment

File: packages/synapse-react-client/src/utils/functions/SanitizeHtmlUtils.test.ts (hunk 4 above)
Comment (nickgros) [reply packages/synapse-react-client/src/utils/functions/SanitizeHtmlUtils.test.ts:23]: yes

Comment (nickgros) [inline packages/synapse-react-client/src/utils/functions/SanitizeHtmlUtils.test.ts:23]:
```suggestion
    expect(span.getAttribute('onclick')).toBeNull()
```

**Pull request merged**